view/statistics-flow-certificates.js

```diff
@@ -0,0 +1,331 @@
+								return selected ? 'selected' : null;
+							})
+						}, ServiceType.prototype.label);
+					})
+				)),
+			div({ class: 'users-table-filter-bar-status' },
+				select(
+					{ id: 'certificate-select', name: 'certificate' },
+					option({ value: '', selected: location.query.get('certificate').map(function (value) {
+						return value ? null : 'selected';
+					}) }, _("All certificates")),
+					list(Object.keys(certificates), function (certificateKey) {
+						return option({
+							id: 'certificate-' + certificateKey,
+							value: certificateKey,
+							selected: location.query.get('certificate').map(function (value) {
+								var selected = (certificateKey ? (value === certificateKey) : (value == null));
+								return selected ? 'selected' : null;
+							})
+						}, certificates[certificateKey].label);
+					}),
+					legacy('selectMatch', 'service-select', serviceToCertLegacyMatch)
+				)),
+			div(
+				{ class: 'users-table-filter-bar-status' },
+				label({ for: 'date-from-input' }, _("Date from"), ":"),
+				input({ id: 'date-from-input', type: 'date',
+					name: 'dateFrom', value: location.query.get('dateFrom').map(function (dateFrom) {
+					var now = new db.Date();
+					now.setDate(now.getDate() - 7);
+					return dateFrom || now.toISOString().slice(0, 10);
+				}) })
+			),
+			div(
+				{ class: 'users-table-filter-bar-status' },
+				label({ for: 'date-to-input' }, _("Date to"), ":"),
+				input({ id: 'date-to-input', type: 'date',
+					name: 'dateTo', value: location.query.get('dateTo') })
+			),
+			div({ class: "input", id: "mode-selection" },
+				div({ class: "inline-button-radio" },
+					list(modes, function (mode) {
+						label(
+							input({
+								type: "radio",
+								name: "mode",
+								value: mode.key,
+								checked: location.query.get('mode').map(function (value) {
+									if (!value) value = 'daily';
+									var checked = (mode.key ? (value === mode.key) : (value == null));
+									return checked ? 'checked' : null;
+								})
+							}),
+							mode.label
+						);
+					})),
+				script((function (id, classMap) {
+					var current, radio, radios;
+					var onChange = function () {
+						var nu, i;
+						for (i = 0; (radio = radios[i]); ++i) {
+							if (radio.checked) {
+								nu = radio;
+								break;
+							}
+						}
+							if (nu === current) return;
+							if (current) current.parentNode.removeClass('success');
+							if (nu) $(nu.parentNode).addClass('success');
+						current = nu;
+					};
+					setTimeout(function self() {
+						var container = $(id);
+						if (!container) {
+							setTimeout(self, 1000);
+							return;
+						}
+						radios = container.getElementsByTagName('input');
+						container.addEvent('change', function () {
+							setTimeout(onChange, 0);
+						});
+						container.addEvent('click', function () {
+							setTimeout(onChange, 0);
+						});
+						onChange();
+					}, 0);
+				}("mode-selection")))),
```

Comment:
What is this `script` doing here? No chance to reuse (or refactor for reuse) what we have made for button group?